packages/theme/src/cli/services/package.test.ts

```diff
@@ -0,0 +1,170 @@
+    await inTemporaryDirectory(async (tmpDir) => {
+      // Given
+      const inputDirectory = joinPath(tmpDir, 'theme')
+      await mkdir(inputDirectory)
+      const themeRelativePaths = [
+        'assets/base.css',
+        'layout/theme.liquid',
+        'config/settings_schema.json',
+        'release-notes.md',
+      ]
+      await createFiles(themeRelativePaths, inputDirectory)
+      await createSettingsSchema('[{"name": "theme_info", "theme_name": "Dawn"}]', inputDirectory)
+
+      // When
+      await packageTheme(inputDirectory)
+
+      // Then
+      const expectedOutputZipPath = joinPath(inputDirectory, 'Dawn.zip')
+      await expect(fileExists(expectedOutputZipPath)).resolves.toBeTruthy()
+
+      const archiveEntries = await readArchiveFiles(expectedOutputZipPath)
+      expect(themeRelativePaths.sort()).toEqual(archiveEntries.sort())
+
+      expect(renderSuccess).toBeCalledWith({
+        headline: `Your local theme was packaged in "${expectedOutputZipPath}".`,
+      })
+    })
+  })
+
+  it('abort if config/settings_schema.json is missing', async () => {
+    await inTemporaryDirectory(async (tmpDir) => {
+      // Given
+      const inputDirectory = joinPath(tmpDir, 'theme')
+      await mkdir(inputDirectory)
+
+      await expect(async () => {
+        // When
+        await packageTheme(inputDirectory)
+
+        // Then
+      }).rejects.toThrowError(/Provide a config\/settings_schema.json to package your theme./)
+
+      expect(renderSuccess).not.toBeCalled()
+    })
+  })
+
+  it('abort if theme name missing from config/settings_schema.json', async () => {
+    await inTemporaryDirectory(async (tmpDir) => {
+      // Given
+      const inputDirectory = joinPath(tmpDir, 'theme')
+      await mkdir(inputDirectory)
+      const themeRelativePaths = ['config/settings_schema.json']
+      await createFiles(themeRelativePaths, inputDirectory)
+      await createSettingsSchema('[{"name": "theme_info"}]', inputDirectory)
+
+      await expect(async () => {
+        // When
+        await packageTheme(inputDirectory)
+
+        // Then
+      }).rejects.toThrowError(/Provide a theme_info.theme_name configuration in config\/settings_schema.json/)
+
+      expect(renderSuccess).not.toBeCalled()
+    })
+  })
+})
+
+async function createFiles(structure: string[], directory: string) {
+  for (const fileRelativePath of structure) {
+    const filePath = joinPath(directory, fileRelativePath)
+    // eslint-disable-next-line no-await-in-loop
+    await mkdir(dirname(filePath))
+    // eslint-disable-next-line no-await-in-loop
+    await touchFile(filePath)
+  }
+}
+
+async function readArchiveFiles(zipPath: string) {
+  await expect(fileExists(zipPath)).resolves.toBeTruthy()
+  // eslint-disable-next-line @babel/new-cap
+  const archive = new StreamZip.async({file: zipPath})
+  const archiveEntries = Object.keys(await archive.entries())
+  await archive.close()
+
+  return archiveEntries
+}
```

Comment:
these functions are also repeated in `archiver.integration.test.ts`, not sure if there's a concept of test helper functions in the cli